include/onnxruntime/core/common/common.h

```diff
@@ -95,6 +99,76 @@ void LogRuntimeError(uint32_t session_id, const common::Status& status, const ch
 #define ORT_WHERE_WITH_STACK \
   ::onnxruntime::CodeLocation(__FILE__, __LINE__, __PRETTY_FUNCTION__, ::onnxruntime::GetStackTrace())
 
+#ifdef ORT_NO_EXCEPTIONS
+
+#define ORT_TRY if (true)
+#define ORT_CATCH(x) else if (false)
+#define ORT_RETHROW
+
+// In order to ignore the catch statement when a specific exception (not ... ) is caught and referred
+// in the body of the catch statements, it is necessary to wrap the body of the catch statement into
+// a lambda function. otherwise the exception referred will be undefined and cause build break
+#define ORT_HANDLE_EXCEPTION(func)
+
+// TODO, consider changing the output of the error message from std::cerr to logging when the
+// exceptions are disabled, since using std::cerr might increase binary size, and std::cerr output
+// might not be easily accesible on some systems such as mobile
+
+// Throw an exception with optional message.
+// NOTE: The arguments get streamed into a string via ostringstream::operator<<
+// DO NOT use a printf format string, as that will not work as you expect.
+#define ORT_THROW(...)                                                                       \
+  do {                                                                                       \
+    std::cerr << ::onnxruntime::OnnxRuntimeException(ORT_WHERE_WITH_STACK,                   \
+                                                     ::onnxruntime::MakeString(__VA_ARGS__)) \
+                     .what()                                                                 \
+              << std::endl;                                                                  \
+    abort();                                                                                 \
+  } while (false)
+
+// Just in order to mark things as not implemented. Do not use in final code.
```

Comment:
Should this be marked as a TODO?

Reply:
Oh, this one is copied from its definition below on line 178


---
In reply to: [478225922](https://github.com/microsoft/onnxruntime/pull/4894#discussion_r478225922) [](ancestors = 478225922)